src/main/java/com/orbitz/consul/Consul.java

```diff
@@ -46,11 +47,12 @@
      */
     private Consul(String url, ClientBuilder builder, ObjectMapper mapper) {
 
-        JacksonJaxbJsonProvider provider = new JacksonJaxbJsonProvider();
-        provider.setMapper(mapper);
-
-        Client client = builder
-                .register(provider)
+        if (!builder.getConfiguration().getClasses().stream()
+                .filter(c -> JacksonJaxbJsonProvider.class.isAssignableFrom(c)).findAny().isPresent()) {
```

Comment:
I'm a little new to Gradle, couldn't figure out if you were using Java 8 or not. Guessing you probably aren't. Could rewrite this in Java 6/7 form if needed